### PR TITLE
Add Create product

### PR DESCRIPTION
**Pull request by enahomurphy**
## Summary 
This ticket adds the ability for shop owner to create new product

### CODE REVIEW

File: app/db/migrations/20171016091057-create-product.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ module.exports = {
         type: Sequelize.DECIMAL
       },
       category_id: {
-        type: Sequelize.INTEGER
+        type: Sequelize.STRING
```
Comment (tomily1) [inline app/db/migrations/20171016091057-create-product.js:17]: is this going to be a UUID? or what is the reason for changing it?

File: app/db/migrations/20171016091057-create-product.js [hunk 2]
```diff
@@ -55,10 +52,14 @@ module.exports = {
       shop_id: {
         type: Sequelize.INTEGER
       },
-      brands_id: {
+      downloadable: {
+        type: Sequelize.BOOLEAN,
+        defaultValue: false
+      },
+      brand_id: {
```
Comment (tomily1) [inline app/db/migrations/20171016091057-create-product.js:59]: I think uniformity is important, if we are going to make use of `UUID` or `Integer` we should make use of it throughout the ids.

File: app/db/migrations/20171016104906-create-product-assets.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ module.exports = {
       product_id: {
         type: Sequelize.INTEGER,
```
Comment (tomily1) [inline app/db/migrations/20171016104906-create-product-assets.js:20]: same as above

File: app/db/models/advanceProductPrice.js [hunk 4]
```diff
@@ -60,6 +42,7 @@ module.exports = (sequelize, DataTypes) => {
       }
     }
   }, {
+    tableName: 'Advance_product_price',
```
Comment (tomily1) [inline app/db/models/advanceProductPrice.js:45]: advance_product_price 😄

File: routes.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      message: err.statusText,
+      errors: err.errors.map(error => ({
+        field: error.field.join(', '),
+        message: error.messages.join(', '),
+        location: error.location,
+      }))
+    });
+  } else if (err.name === 'SequelizeValidationError') {
+    return res.status(422).json({
+      status: 422,
```
Comment (tomily1) [inline routes.js:42]: good stuff

**Review by tomily1 — APPROVED** [review]
how far? I go approve am but you gats make the changes sir😄

File: app/db/migrations/20171016091057-create-product.js (hunk 1 above)
Comment (enahomurphy) [reply app/db/migrations/20171016091057-create-product.js:17]: This has nothing to do with UUID, If you check the models you will have a broader sense of what I'm trying to achieve. a product can belong to multiple categories, so basically It require d that the user send an array of category ID. In the model, All I did was to convert that array to a comma-separated string before saving. If there is any performance gain in using UUID then we can switch to that.

File: app/db/migrations/20171016091057-create-product.js (hunk 2 above)
Comment (enahomurphy) [reply app/db/migrations/20171016091057-create-product.js:59]: There is uniformity, integer is what we're using

File: routes.js (hunk 5 above)
Comment (enahomurphy) [reply routes.js:42]: Thanks

**Pull request merged**